runtime/onert/core/src/backend/controlflow/kernel/PermuteLayer.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PermuteLayer.h"
+
+#include "exec/ShapeConverter.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace controlflow
+{
+namespace kernel
+{
+
+void PermuteLayer::run()
+{
+  assert(_src_tensors.size() == _dst_tensors.size());
+  // check if output is not dynamic
+  for (size_t i = 0; i < _src_tensors.size(); ++i)
+  {
+    auto dst_tensor = _dst_tensors.at(i);
+    auto src_tensor = _src_tensors.at(i);
+    if (dst_tensor->is_dynamic())
+    {
```

Comment:
Unfortunately, `FunctionSequenceForDynamicBackend` cannot support inferring shape of controlflow operations. There are some reasons as follows:
1. In case of `If`/`While` operations,  there is no way to infer output shape with using only input shape.
2. The operations access destination tensors of other executor. But `FunctionSequenceForDynamicBackend` cannot support to access tensor and tensor manager of other executor.
3. `Permute` operation can access tensors of other backends.
4. `If`/`While` operations use `PermuteLayer` to copy tensors between other executors.

Reply:
I just added comments about it.